connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -299,103 +304,115 @@ public StatusBackingStore statusBackingStore() {
-            ConfigInfos producerConfigInfos = null;
-            ConfigInfos consumerConfigInfos = null;
-            ConfigInfos adminConfigInfos = null;
-            if (connectorType.equals(org.apache.kafka.connect.health.ConnectorType.SOURCE)) {
-                producerConfigInfos = validateClientOverrides(connName,
-                                                              ConnectorConfig.CONNECTOR_CLIENT_PRODUCER_OVERRIDES_PREFIX,
-                                                              connectorConfig,
-                                                              ProducerConfig.configDef(),
-                                                              connector.getClass(),
-                                                              connectorType,
-                                                              ConnectorClientConfigRequest.ClientType.PRODUCER,
-                                                              connectorClientConfigOverridePolicy);
-                return mergeConfigInfos(connType, configInfos, producerConfigInfos);
-            } else {
-                consumerConfigInfos = validateClientOverrides(connName,
-                                                              ConnectorConfig.CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX,
-                                                              connectorConfig,
-                                                              ProducerConfig.configDef(),
-                                                              connector.getClass(),
-                                                              connectorType,
-                                                              ConnectorClientConfigRequest.ClientType.CONSUMER,
-                                                              connectorClientConfigOverridePolicy);
-                // check if topic for dead letter queue exists
-                String topic = connectorProps.get(SinkConnectorConfig.DLQ_TOPIC_NAME_CONFIG);
-                if (topic != null && !topic.isEmpty()) {
-                    adminConfigInfos = validateClientOverrides(connName,
-                                                               ConnectorConfig.CONNECTOR_CLIENT_ADMIN_OVERRIDES_PREFIX,
-                                                               connectorConfig,
-                                                               ProducerConfig.configDef(),
-                                                               connector.getClass(),
-                                                               connectorType,
-                                                               ConnectorClientConfigRequest.ClientType.ADMIN,
-                                                               connectorClientConfigOverridePolicy);
+    public void validateConnectorConfig(Map<String, String> connectorProps, Callback<ConfigInfos> callback) {
+        connectorExecutor.submit(new Runnable() {
```

Comment:
Nit: maybe put the runnable logic as a separate method so that far fewer lines are changed.

Reply:
Ack, addressed within the `AbstractHerder` class and will see if similar changes can be made elsewhere.